source/dlangbot/database.d

```diff
@@ -0,0 +1,54 @@
+/**
+Implements a database to keep track of the points scored
+by authors that fix bugs. The table has the following
+columns:
+
+Github account (Primary key)
+Total points
+*/
+module dlangbot.database;
+
+import ae.sys.database : Database;
+
+private Database _db;
+enum table_schema = q"SQL
+CREATE TABLE [LEADERBOARD] (
+[AUTHOR]  TEXT PRIMARY KEY NOT NULL,
+[POINTS]  INTEGER
+);
+SQL";
+
+/**
+Checks whether a database file already exists.
+If that is the case, it returns the database,
+otherwise it creates it.
+*/
+Database initDatabase(string databasePath)
+{
+    import std.file : exists;
+    import ae.sys.file : ensurePathExists;
+
+    if (databasePath.exists)
+        return Database(databasePath);
+
+    ensurePathExists(databasePath);
+    return Database(databasePath, [table_schema,]);
+}
+
+void updateScoreBoard(string author, uint points, Database db = _db)
+{
+    uint authorExistingPoints = 0;
+    foreach(int _points; db.stmt!"SELECT [POINTS] FROM [LEADERBOARD] WHERE [AUTHOR]=?".iterate(author))
+        authorExistingPoints = _points;
+
+    if (authorExistingPoints == 0)
+        db.stmt!"INSERT INTO [LEADERBOARD] ([AUTHOR], [POINTS]) VALUES (?, ?)".exec(author, points);
+    else
+        db.stmt!"UPDATE [LEADERBOARD] SET [POINTS]=? WHERE [AUTHOR]=?".exec(authorExistingPoints+points, author);
```

Comment:
I suggest the following simplification:
```suggestion
    db.stmt!"
        INSERT OR REPLACE INTO [LEADERBOARD] ([AUTHOR], [POINTS]) VALUES (?1,
            COALESCE((SELECT [POINTS] FROM [LEADERBOARD] WHERE [AUTHOR] = ?1), 0) + ?2
        )".exec(author, points);
```

`EXPLAIN QUERY PLAN` shows that SQLite performs the update in a single lookup.